src/ConnectedMode/Migration/ConnectedModeMigration.cs

```diff
@@ -188,11 +188,20 @@ private System.Threading.Tasks.Task<string> GetFileContentAsync(string filePath)
         private async Task MakeLegacyFileChangesAsync(LegacySettings legacySettings, ChangedFiles changedFiles,
             IProgress<MigrationProgress> progress)
         {
-            // Note: no files have been changed yet. Now we are going to start making changes
-            // to the user's projects and deleting files that might be under source control...
-            await SaveChangedFilesAsync(changedFiles, progress);
+            try
+            {
+                await fileSystem.BeginChangeBatchAsync();
 
-            await DeleteSonarLintFolderAsync(legacySettings, progress);
+                // Note: no files have been changed yet. Now we are going to start making changes
+                // to the user's projects and deleting files that might be under source control...
+                await SaveChangedFilesAsync(changedFiles, progress);
+
+                await DeleteSonarLintFolderAsync(legacySettings, progress);
+            }
+            finally
+            {
+                await fileSystem.EndChangeBatchAsync();
+            }
```

Comment:
Add the batch operations to the interface. They don't do anything at this point.
We've closing the batch in a `finally` block to make sure it happens even if there are exceptions.